lib/protocol/protocol.go

```diff
@@ -363,25 +370,44 @@ func (c *rawConnection) ping() bool {
 	return c.send(&Ping{}, nil)
 }
 
-func (c *rawConnection) readerLoop() (err error) {
-	defer c.wg.Done()
+type errorMessage struct {
```

Comment:
readResponse? This feels like an actual error message not the conjunction of two types independently.

Reply:
Ou, I see how that's confusing now (I went `asyncMessage` -> `errorMessage` in my head). `readResponse` also is a bit ambiguous, as there is a `Response` message that gets read. Maybe `messageWithError`?

Reply:
Yeah